app/lib/logstash_logging.rb

```diff
@@ -1,49 +1,76 @@
+    # Log destination: log to STDOUT, plus to a remote Logstash server, if required
+    logstash_logger = \
+      if ENV['LOGSTASH_REMOTE'] == 'true'
+        LogStashLogger.new(
+          type: :multi_delegator,
+          outputs: [
+            { type: :stdout }, # good practice for Docker containers
+            {
+              type: :tcp,
+              host: ENV.fetch('LOGSTASH_HOST'),
+              port: ENV.fetch('LOGSTASH_PORT'),
+              ssl_enable: ENV.fetch('LOGSTASH_SSL') == 'true',
+            },
+          ],
+        )
+      else
+        LogStashLogger.new(type: :stdout)
+      end
 
-    # Make stdout_logger the standard Rails logger
-    stdout_logger = ActiveSupport::Logger.new(STDOUT)
-    stdout_logger.formatter = config.log_formatter
-    config.logger = ActiveSupport::TaggedLogging.new(stdout_logger)
+    # Make logstash_logger the default logger for Rails, Sidekiq and clockwork
+    rails_config.logger = logstash_logger
+    Sidekiq.logger = logstash_logger
+    Clockwork.configure { |config| config[:logger] = logstash_logger }
+  end
 
-    # Use lograge for one log line per request, in json format for Logstash
-    config.lograge.enabled = true # lograge uses one log line per request
-    config.lograge.base_controller_class = [
+  def self.lograge_config(rails_config)
+    rails_config.lograge.enabled = true # lograge uses one log line per request
+    rails_config.lograge.base_controller_class = [
       'ActionController::Base',
       'ActionController::API', # API controllers inherit from ActionController::API
     ]
-    config.lograge.formatter = Lograge::Formatters::Logstash.new
-
-    # Add query params to the log format - lograge usually ignores them
-    config.lograge.custom_options = lambda do |event|
-      ignore_params = %w(candidate authenticity_token)
-      {
-        domain: DOMAIN_FOR_LOGS,
-        params: event.payload[:params].except(*ignore_params),
-        candidate_id: event.payload[:candidate_id],
-        vendor_api_token_id: event.payload[:vendor_api_token_id],
-        provider_id: event.payload[:provider_id],
-        exception: event.payload[:exception], # ["ExceptionClass", "the message"]
-      }
+    # Add exception to the lograge log format
+    rails_config.lograge.custom_options = lambda do |event|
+      { exception: event.payload[:exception] } # ["ExceptionClass", "the message"]
     end
+    rails_config.lograge.formatter = Lograge::Formatters::Logstash.new
+  end
 
-    if ENV['LOGSTASH_REMOTE'] == 'true'
-      tcp_logger = LogStashLogger.new(type: :tcp,
-                                      host: ENV.fetch('LOGSTASH_HOST'),
-                                      port: ENV.fetch('LOGSTASH_PORT'),
-                                      ssl_enable: ENV.fetch('LOGSTASH_SSL') == 'true')
+  def self.add_identity_fields(event)
+    identity_hash = RequestLocals.fetch(:identity) { nil }
+    identity_hash.each { |key, val| event[key] = val } if identity_hash
```

Comment:
Does `event.merge(identity_hash)` not do the trick?

Reply:
No, it's not a hash, it's a LogStash::Event object that provides a hash-like interface.